coruscant/src/components/user-settings-tabs/AccountSettings.js

```diff
@@ -1,51 +1,201 @@
+import BasicSpinner from '../../components/BasicSpinner'
+import { checkGithubToken, deleteGithubToken, checkHerokuToken, deleteHerokuToken, setGithubState, setHerokuState } from '../../actions/session'
 
 const styles = {
   container: {
     padding: Theme.padding.tiny
-  },
-  bottomActionSpacer: {
-    height: Theme.padding.large
-  },
-  hr: {
-    height: 1,
-    border: 0,
-    borderTop: `1px solid ${Theme.palette.primary3Color}`,
-    marginTop: Theme.padding.small,
-    marginBottom: Theme.padding.small
   }
 }
 
+const mapStateToProps = state => ({
+  session: state.session
+})
+
+const mapDispatchToProps = {
+  checkGithubToken,
+  deleteGithubToken,
+  checkHerokuToken,
+  deleteHerokuToken,
+  setGithubState,
+  setHerokuState
+}
+
+@connect(mapStateToProps, mapDispatchToProps)
 class AccountSettings extends Component {
-  // eslint-disable-next-line class-methods-use-this
-  renderAccountInfo(account) {
-    const label = `Connect to ${account.name}`
-    const info = (account.linked)
-      ? (
+  static getUnguessableString() {
+    return Math.random().toString(36).substring(2, 15)
+      + Math.random().toString(36).substring(2, 15)
+    // return '4'
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      github: 'spin',
+      heroku: 'spin'
+    }
+  }
+
+  componentDidMount() {
+    this.checkGithub()
+    this.checkHeroku()
+  }
+
+  async checkGithub() {
+    const {
+      checkGithubToken,
+      session
+    } = this.props
+    const { result } = await checkGithubToken(session.authToken)
+    if (result) {
+      this.setState({ github: result.isGithubAuthenticated })
+    } else {
+      // TODO
+      // This should actually set github to spin,
+      // but the backend is throwing an error when there is no token.
+      this.setState({ github: false })
+    }
+  }
+
+  async checkHeroku() {
+    const {
+      checkHerokuToken,
+      session
+    } = this.props
+    const { result } = await checkHerokuToken(session.authToken)
+    if (result) {
+      this.setState({ heroku: result.isHerokuAuthenticated })
+    } else {
+      this.setState({ heroku: 'spin' })
+    }
+  }
+
+  async removeGithub() {
+    const {
+      deleteGithubToken,
+      session
+    } = this.props
+    const { error } = await deleteGithubToken(session.authToken)
+    if (!error) {
+      this.setState({ github: false })
+    }
+  }
+
+  async removeHeroku() {
+    const {
+      deleteHerokuToken,
+      session
+    } = this.props
+    const { error } = await deleteHerokuToken(session.authToken)
+    if (!error) {
+      this.setState({ heroku: false })
+    }
+  }
+
+  async redirectToGithub() {
+    const state = AccountSettings.getUnguessableString()
+    const {
+      session
+    } = this.props
+
+    saveState({
+      session,
+      githubState: state
+    })
+
+    window.location = `${'https://github.com/login/oauth/authorize?' +
+    'client_id=6b57706945fa18ee0397&' +
```

Comment:
Extract the client_id to a config file

Reply:
Discussed and this will be added to Nathan's firebase PR since configurability was introduced there